pkg/syndesis/action/install.go

```diff
@@ -0,0 +1,96 @@
+package action
+
+import (
+	"errors"
+	"github.com/openshift/api/template/v1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	"github.com/syndesisio/syndesis-operator/pkg/apis/syndesis/v1alpha1"
+	"github.com/syndesisio/syndesis-operator/pkg/openshift/serviceaccount"
+	"github.com/syndesisio/syndesis-operator/pkg/openshift/template"
+	"github.com/syndesisio/syndesis-operator/pkg/util"
+	coreV1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// Install syndesis into the namespace, taking resources from the template
+
+type Install struct {}
+
+
+func (a *Install) CanExecute(syndesis *v1alpha1.Syndesis) bool {
+	return syndesisInstallationStatusIs(syndesis, v1alpha1.SyndesisInstallationStatusInstalling)
+}
+
+func (a *Install) Execute(syndesis *v1alpha1.Syndesis) error {
+
+	logrus.Info("Installing Syndesis resource ", syndesis.Name)
+
+	serviceAccountRes, err := util.LoadKubernetesResourceFromAsset("oauth-client-sa.yaml")
+	if err != nil {
+		return err
+	}
+
+	var saName string
+	if sa, ok := serviceAccountRes.(*coreV1.ServiceAccount); ok {
+		saName = sa.Name
+	} else {
+		return errors.New("Cannot determine service account name")
+	}
+
+	customizeKubernetesResource(serviceAccountRes, syndesis)
```

Comment:
maybe better name `setNamespaceAndOwnerReference` ? (just to make it a bit more specific)